### PR TITLE
generalize transformer to also handle partial returns

### PR DESCRIPTION
**Pull request by 0xbe7a**

### CODE REVIEW

**Comment by 0xbe7a** [comment]
closes #8

File: polarify/main.py [hunk 1]
```diff
@@ -36,41 +36,43 @@ def __init__(self, assignments: dict[str, ast.expr]):
         self.assignments = assignments
 
     @classmethod
-    def inline_expr(cls, expr: ast.expr, assignments: dict[str, ast.expr]) -> ast.expr:
-        return cls(assignments).visit(deepcopy(expr))
+    def inline_expr(cls, expr: ast.expr, assignments: dict[str, ast.expr]) -> ast.expr: # TODO
+        expr = cls(assignments).visit(deepcopy(expr))
+        assert isinstance(expr, ast.expr)
```
Comment (0xbe7a) [inline polarify/main.py:41]: Why is this required?

File: polarify/main.py [hunk 2]
```diff
@@ -36,41 +36,43 @@ def __init__(self, assignments: dict[str, ast.expr]):
         self.assignments = assignments
 
     @classmethod
-    def inline_expr(cls, expr: ast.expr, assignments: dict[str, ast.expr]) -> ast.expr:
-        return cls(assignments).visit(deepcopy(expr))
+    def inline_expr(cls, expr: ast.expr, assignments: dict[str, ast.expr]) -> ast.expr: # TODO
```
Comment (0xbe7a) [inline polarify/main.py:39]: TODO?

File: polarify/main.py (hunk 1 above)
Comment (pavelzw) [reply polarify/main.py:41]: Just a sanity check that it returns the right thing
